library/core/src/num/int_macros.rs

```diff
@@ -183,6 +183,30 @@ macro_rules! int_impl {
             (self as $UnsignedT).trailing_ones()
         }
 
+        /// Returns the bit pattern of `self` reinterpreted as an unsigned integer of the same size.
+        ///
+        /// This is a bit safer than `as` because it wouldn't silently change the size if the code
```

Comment:
```suggestion
        /// This produces the same result as an `as` cast, but ensures that the bit-width remains the same.
```

I don't think "safe" is a good term to use here given `as` casts are equally safe in the unsafe/safe sense. Something like this maybe?

On pointers a similar method has no extra explanation, so maybe just dropping this line is also fine: https://doc.rust-lang.org/nightly/std/primitive.pointer.html#method.cast-1

Reply:
I took the wording from the `cast_const` function of pointers, where it's also not really a safety thing is it? Maybe it ought to be changed there then.